### ISSUE TITLE
onChange returns the string representation instead of the object.

### ISSUE THREAD

**Issue by rayhooker** [downshift-js/downshift#1228]
- `downshift` version:  6.1.0
- `node` version: 12.18.4
- `npm` (or `yarn`) version: 6.14.11

**Relevant code or config**

I am basically using the code from the material-ui:
https://v3.material-ui.com/demos/autocomplete/

A running version is as follows:
https://stackblitz.com/edit/urufcn?file=package.json
 

**What you did**:
Basically I created a list of objects where the "label" was to be displayed for selection but with a "key".   I want to return the key.  

**What happened**:
 Basically it only returns the "value" for the selected item rather than the object to "onChange".

**Reproduction repository**:
 
https://stackblitz.com/edit/urufcn?file=package.json

**Problem description**:
It does not return the object to "onChange"

**Suggested solution**:

**Comment by rayhooker** [comment]
I actually noticed the code redefined the itemprops in the input parameters to renderSuggestion:

itemProps: getItemProps({ item: suggestion.label })

That clearly translated the selected item from object to string.  I just removed it and added an itemToString to select out the label.  This allowed to return the object but render to a string.

**Comment by silviuaavram** [comment]
You should also consider using `useCombobox` instead of `Downshift`.
